nexusutils/dataconverter/writer.py

```diff
@@ -202,22 +207,27 @@ def __nxdl_to_attrs(self, path: str = '/') -> dict:
 
         # Fetch values for required attributes requested by the NXDL
         for attr_name in elem.findall(f"{self.nxs_namespace}attribute"):
-            elem.attrib[attr_name.get('name')] = self.data[f"{path}/@{attr_name.get('name')}"] or ''
+            key = f"{path}/@{attr_name.get('name')}"
+            if key in self.data:
+                elem.attrib[attr_name.get('name')] = self.data[key]
+            else:
+                elem.attrib[attr_name.get('name')] = ''
 
         return elem.attrib
 
     def ensure_and_get_parent_node(self, path: str, undocumented_paths) -> h5py.Group:
         """Returns the parent if it exists for a given path else creates the parent group."""
         parent_path = path[0:path.rindex('/')] or '/'
         parent_path_hdf5 = helpers.convert_data_dict_path_to_hdf5_path(parent_path)
-        parent_undocumented_paths = [path[0:path.rindex("/")] or "/" for path in undocumented_paths]
         if not does_path_exist(parent_path, self.output_nexus):
-            parent = self.ensure_and_get_parent_node(parent_path, parent_undocumented_paths)
+            parent = self.ensure_and_get_parent_node(parent_path, undocumented_paths)
             grp = parent.create_group(parent_path_hdf5)
-            if path not in undocumented_paths:
```

Comment:
This cannot be removed as the we want to make sure if there is a path that is not documented, we handle it accordingly. In this case, we will raise an exception instead of skipping over undocumented paths. There is a function in helper.py that checks and places undocumented paths to their respective place.

Reply:
actually, this check is now replaced by the __nxdl_to_attrs